libs/wingc/src/type_check.rs

```diff
@@ -1356,33 +1356,40 @@ impl<'a> TypeChecker<'a> {
 
 				// Lookup the class's type in the env
 				let type_ = self.resolve_type_annotation(class, env);
-				let (class_env, class_symbol) = match &*type_ {
-					Type::Class(ref class) => {
-						if class.phase == Phase::Independent || env.phase == class.phase {
-							(&class.env, &class.name)
-						} else {
-							self.spanned_error(
-								exp,
-								format!(
-									"Cannot create {} class \"{}\" in {} phase",
-									class.phase, class.name, env.phase
-								),
-							);
-							return self.types.error();
+				let (class_env, class_symbol) =
+					match &*type_ {
+						Type::Class(ref class) => {
+							if class.phase == Phase::Independent || env.phase == class.phase {
+								(&class.env, &class.name)
+							} else {
+								self.spanned_error(
+									exp,
+									format!(
+										"Cannot create {} class \"{}\" in {} phase",
+										class.phase, class.name, env.phase
+									),
+								);
+								return self.types.error();
+							}
 						}
-					}
-					t => {
-						if matches!(t, Type::Anything) {
-							return self.types.anything();
-						} else {
-							self.spanned_error(
+						t => {
+							if matches!(t, Type::Anything) {
+								return self.types.anything();
+							} else if matches!(t, Type::Struct(_)) {
+								self.spanned_error(
 								class,
-								format!("Cannot instantiate type \"{}\" because it is not a class", type_),
+								format!("Cannot instantiate type \"{}\" because it is a struct and not a class. Use struct instantiation instead.", type_),
 							);
-							return self.types.error();
+								return self.types.error();
+							} else {
+								self.spanned_error(
```

Comment:
@fynnfluegge - what I mean is - is there a way to test that `else` code? What's the wing code that produces this error message?

Reply:
@staycoolcall911 Alright got it, will add this kind of test case ✅

Reply:
Makes sense to me to keep it in.